lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -635,6 +636,18 @@ def edit_config(self, candidate=None, commit=True, replace=None, comment=None):
 
         return resp
 
+    def set_module_specific_options(self):
+        """Set module-specific options; e.g. minimum timeout requirements.
+        """
+        if self._module._name == 'nxos_gir':
+            gir_timeout = 200
+            conn = self._connection
+            persistent_command_timeout = conn.get_option('persistent_command_timeout')
+            if persistent_command_timeout < gir_timeout:
+                conn.set_option('persistent_command_timeout', gir_timeout)
+                msg = "timeout value extended to %ss for nxos_gir." % gir_timeout
+                self._module.warn(msg)
+
```

Comment:
Note that this method exists in the `httpapi` class so it doesn't need to check for other transport states.

Reply:
The logic to set command timeout can be moved to nxos action plugin
```
        conn = Connection(self._connection.socket_path)
        timeout = conn.get_option('persistent_command_timeout')
        conn.set_options(direct={'persistent_command_timeout': 241})
```